### PR TITLE
Fix: [node-cleanup] Check node label to know if pv is linked to a deleted node

### PR DESCRIPTION
**Pull request by Fricounet**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

The current implementation of the node cleanup controller has a bug where if the `kubernetes.io/hostname` in the affinity of the PV is not the same as the `node.Name`, all the local PV and PVC will be considered as being linked to a deleted node and be deleted. We have this situation on AWS clusters where the label is `kubernetes.io/hostname: ip-10-10-10-10` but the node name is `ip-10-10-10-10.ec2.internal`. 
This PR changes how the controller look for the node to now rely on the label on the node rather than  the node name. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note

```

### CODE REVIEW

**Comment by Fricounet** [comment]
@andyzhangx do you think you could take a look at this PR when you have some time?

File: pkg/common/common.go [hunk 1]
```diff
@@ -488,12 +489,17 @@ func GetVolumeMode(volUtil util.VolumeUtil, fullPath string) (v1.PersistentVolum
 }
 
 // NodeExists checks to see if a Node exists in the Indexer of a NodeLister.
-func NodeExists(nodeLister corelisters.NodeLister, nodeName string) (bool, error) {
-	_, err := nodeLister.Get(nodeName)
-	if errors.IsNotFound(err) {
-		return false, nil
+// It uses the well known label `kubernetes.io/hostname` to find the Node.
+func NodeExists(nodeLister corelisters.NodeLister, nodeLabelValue string) (bool, error) {
+	req, err := labels.NewRequirement(NodeLabelKey, selection.Equals, []string{nodeLabelValue})
```
Comment (andyzhangx) [inline pkg/common/common.go:494]: can you use the original method `_, err := nodeLister.Get(nodeName)` and if not found, use label as fallback, that would be safer, thx

Comment (Fricounet) [reply pkg/common/common.go]: good point. I've modified the code to keep the original get

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**